File: www/faceDetection.js
/*
 * Copyright 2021 TripodWorks
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var exec = require('cordova/exec');
var faceDetection = function () {};

faceDetection.prototype.start = function(param, success, error) {
  exec(success, error, "faceDetection", "start", [param]);
};

faceDetection.prototype.stop = function(success, error) {
  exec(success, error, "faceDetection", "stop", []);
};

faceDetection.prototype.takePicture = function(param, success, error) {
  exec(success, error, "faceDetection", "takePicture", [param]);
};

faceDetection.prototype.startX = function(param, success, error) {
  exec(success, error, "faceDetection", "startX", [param]);
};

faceDetection.prototype.stopX = function(success, error) {
  exec(success, error, "faceDetection", "stopX", []);
};

var FaceDetection = new faceDetection();
module.exports = FaceDetection;

File: src/ios/faceDetection.m
/*
 * Copyright 2021 TripodWorks
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#import <Cordova/CDV.h>
#import <Cordova/CDVPlugin.h>
#import <Cordova/CDVInvokedUrlCommand.h>
#import <GLKit/GLKit.h>
#import "faceDetection.h"

@implementation faceDetection

-(void) pluginInitialize{
  // start as transparent
  self.webView.opaque = NO;
  self.webView.backgroundColor = [UIColor clearColor];
}

- (void)start:(CDVInvokedUrlCommand*)command {
    NSLog(@"start call");
    CDVPluginResult* pluginResult = nil;

    if (self.sessionManager != nil) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Camera already started!"];
        [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
        return;
    }

    NSData * data = [command.arguments objectAtIndex:0];
    NSError *error = nil;
    NSDictionary object = [NSJSONSerialization
                      JSONObjectWithData:data
                      options:0
                      error:&error];




    if (echo != nil && [echo length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void) stop:(CDVInvokedUrlCommand*)command {
    NSLog(@"stop call");
    CDVPluginResult *pluginResult;

    if(self.sessionManager != nil) {
        [self.cameraRenderController.view removeFromSuperview];
        [self.cameraRenderController removeFromParentViewController];

        self.cameraRenderController = nil;
        self.sessionManager = nil;

        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
    }
    else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Camera not started"];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}


@end
